src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -71,20 +72,31 @@ public sealed partial class TextLoader
             char separator = '\t', bool allowQuotedStrings = true,
             bool supportSparse = true, bool trimWhitespace = false)
         {
-            var fields = typeof(TInput).GetFields();
-            Arguments.Column = new TextLoaderColumn[fields.Length];
-            for (int index = 0; index < fields.Length; index++)
+            var userType = typeof(TInput);
+
+            var fieldInfos = userType.GetFields();
+
+            var propertyInfos =
+                userType
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Where(x => x.CanRead && x.CanWrite && x.GetIndexParameters().Length == 0)
+                .ToArray();
```

Comment:
Shouldn't need to `ToArray` here, when you are just going to concat it with `fieldInfos` and then `ToArray` that as well.

Reply:
I'll remove it, thanks